src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -1201,31 +1201,50 @@ public static SqlConnectionStringBuilder CreateConnectionStringBuilder(Connectio
             ChangeDatabaseParams changeDatabaseParams,
             RequestContext<bool> requestContext)
         {
-            await requestContext.SendResult(ChangeConnectionDatabaseContext(changeDatabaseParams.OwnerUri, changeDatabaseParams.NewDatabase));
+            await requestContext.SendResult(ChangeConnectionDatabaseContext(changeDatabaseParams.OwnerUri, changeDatabaseParams.NewDatabase, true));
         }
 
         /// <summary>
         /// Change the database context of a connection.
         /// </summary>
         /// <param name="ownerUri">URI of the owner of the connection</param>
         /// <param name="newDatabaseName">Name of the database to change the connection to</param>
-        public bool ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseName)
+        public bool ChangeConnectionDatabaseContext(string ownerUri, string newDatabaseName, bool force = false)
         {
             ConnectionInfo info;
             if (TryFindConnection(ownerUri, out info))
             {
                 try
                 {
-                    foreach (DbConnection connection in info.AllConnections)
+                    info.ConnectionDetails.DatabaseName = newDatabaseName;
+
+                    foreach (string key in info.AllConnectionTypes)
                     {
-                        if (connection.State == ConnectionState.Open)
+                        DbConnection conn;
+                        info.TryGetConnection(key, out conn);
+                        if (conn != null && conn.State == ConnectionState.Open)
                         {
-                            connection.ChangeDatabase(newDatabaseName);
+                            if (info.IsCloud && force)
+                            {
+                                conn.Close();
+                                conn.Dispose();
+                                info.RemoveConnection(key);
+                                
+                                string connectionString = BuildConnectionString(info.ConnectionDetails);
+
+                                // create a sql connection instance
+                                DbConnection connection = info.Factory.CreateSqlConnection(connectionString);
+                                connection.Open();
+                                info.AddConnection(key, connection);
+                            }
+                            else
+                            {
+                                conn.ChangeDatabase(newDatabaseName);

```

Comment:
Is there ever a case where ChangeDatabase will actually work for IsCloud?
- I think we should probably always use this codepath for cloud, since the ChangeDatabase is failing
- I don't see this having a net impact on the Query execution codepath so we might not need to have a `force` option. The reason is that `USE DATABASE` will not work, so it will never be executed

Reply:
The current behavior is if you use `use database` in a query again cloud you get an error saying you can't do that. I wanted to keep that behavior for query execution (since it is what ssms does) so thats why I added the force method.

Reply:
What I'm saying is that the code path won't get called - what really happens is that the ConnectionType.Query connection is already changed to be the new DB, and this code path just changes the others. In any case, not a blocker.